convert-hf-to-gguf.py

```diff
@@ -28,6 +30,46 @@
 logger = logging.getLogger("hf-to-gguf")
 
 
+@dataclass
+class Metadata:
```

Comment:
If you want to support `convert-legacy-llama.py` perhaps it makes sense to refactor that class out, since it seems to be the same as the one there.